typescript/utils/src/altvm.ts

```diff
@@ -0,0 +1,682 @@
+
+  getMerkleRootMultisigIsm(
+    req: ReqMerkleRootMultisigIsm,
+  ): Promise<ResMerkleRootMultisigIsm>;
+
+  getRoutingIsm(req: ReqRoutingIsm): Promise<ResRoutingIsm>;
+
+  getNoopIsm(req: ReqNoopIsm): Promise<ResNoopIsm>;
+
+  getHookType(req: ReqGetHookType): Promise<HookType>;
+
+  getInterchainGasPaymasterHook(
+    req: ReqGetInterchainGasPaymasterHook,
+  ): Promise<ResGetInterchainGasPaymasterHook>;
+
+  getMerkleTreeHook(req: ReqGetMerkleTreeHook): Promise<ResGetMerkleTreeHook>;
+
+  // ### QUERY WARP ###
+
+  getToken(req: ReqGetToken): Promise<ResGetToken>;
+
+  getRemoteRouters(req: ReqGetRemoteRouters): Promise<ResGetRemoteRouters>;
+
+  getBridgedSupply(req: ReqGetBridgedSupply): Promise<bigint>;
+
+  quoteRemoteTransfer(
+    req: ReqQuoteRemoteTransfer,
+  ): Promise<ResQuoteRemoteTransfer>;
+
+  // ### GET CORE TXS ###
+
+  getCreateMailboxTransaction(req: ReqCreateMailbox): Promise<T>;
+
+  getSetDefaultIsmTransaction(req: ReqSetDefaultIsm): Promise<T>;
+
+  getSetDefaultHookTransaction(req: ReqSetDefaultHook): Promise<T>;
+
+  getSetRequiredHookTransaction(req: ReqSetRequiredHook): Promise<T>;
+
+  getSetMailboxOwnerTransaction(req: ReqSetMailboxOwner): Promise<T>;
+
+  getCreateMerkleRootMultisigIsmTransaction(
+    req: ReqCreateMerkleRootMultisigIsm,
+  ): Promise<T>;
+
+  getCreateMessageIdMultisigIsmTransaction(
+    req: ReqCreateMessageIdMultisigIsm,
+  ): Promise<T>;
+
+  getCreateRoutingIsmTransaction(req: ReqCreateRoutingIsm): Promise<T>;
+
+  getSetRoutingIsmRouteTransaction(req: ReqSetRoutingIsmRoute): Promise<T>;
+
+  getRemoveRoutingIsmRouteTransaction(
+    req: ReqRemoveRoutingIsmRoute,
+  ): Promise<T>;
+
+  getSetRoutingIsmOwnerTransaction(req: ReqSetRoutingIsmOwner): Promise<T>;
+
+  getCreateNoopIsmTransaction(req: ReqCreateNoopIsm): Promise<T>;
+
+  getCreateMerkleTreeHookTransaction(req: ReqCreateMerkleTreeHook): Promise<T>;
+
+  getCreateInterchainGasPaymasterHookTransaction(
+    req: ReqCreateInterchainGasPaymasterHook,
+  ): Promise<T>;
+
+  getSetInterchainGasPaymasterHookOwnerTransaction(
+    req: ReqSetInterchainGasPaymasterHookOwner,
+  ): Promise<T>;
+
+  getSetDestinationGasConfigTransaction(
+    req: ReqSetDestinationGasConfig,
+  ): Promise<T>;
+
+  getCreateValidatorAnnounceTransaction(
+    req: ReqCreateValidatorAnnounce,
+  ): Promise<T>;
+
+  // ### GET WARP TXS ###
+
+  getCreateCollateralTokenTransaction(
+    req: ReqCreateCollateralToken,
+  ): Promise<T>;
+
+  getCreateSyntheticTokenTransaction(req: ReqCreateSyntheticToken): Promise<T>;
+
+  getSetTokenOwnerTransaction(req: ReqSetTokenOwner): Promise<T>;
+
+  getSetTokenIsmTransaction(req: ReqSetTokenIsm): Promise<T>;
+
+  getEnrollRemoteRouterTransaction(req: ReqEnrollRemoteRouter): Promise<T>;
+
+  getUnenrollRemoteRouterTransaction(req: ReqUnenrollRemoteRouter): Promise<T>;
+
+  getRemoteTransferTransaction(req: ReqRemoteTransfer): Promise<T>;
+}
+
+export interface ISigner<T, R extends Receipt = any> extends IProvider<T> {
```

Comment:
This could be `R extends Receipt = Receipt` but even better would be to remove the default and propagate correct generic parameters like we did with transactions.

Reply:
propagating the correct generic receipt now